modules/manager/controller_setup.go

```diff
@@ -165,42 +166,57 @@ func (c *ControllerDef) MaybeSetupWithManager(ctx context.Context, mgr ctrl.Mana
 	return c.Controller.SetupWithManager(ctx, mgr)
 }
 
-func setupIndexes(ctx context.Context, mgr manager.Manager, cfg Config) error {
+// SetupCacheIndexes sets up all the cache indexes required by the controllers.
+func SetupCacheIndexes(ctx context.Context, mgr manager.Manager, cfg Config) error {
+	var indexOptions []index.Option
+
 	if cfg.ControlPlaneControllerEnabled || cfg.GatewayControllerEnabled {
-		log.GetLogger(ctx, "ControlPlane", cfg.DevelopmentMode).Info(
-			"creating index",
-			"indexField", index.DataPlaneNameIndex,
-		)
-		if err := index.DataPlaneNameOnControlPlane(ctx, mgr.GetCache()); err != nil {
-			return fmt.Errorf("failed to setup index for DataPlane names on ControlPlane: %w", err)
-		}
+		indexOptions = append(indexOptions, index.OptionsForControlPlane()...)
 		if cfg.KongPluginInstallationControllerEnabled {
-			log.GetLogger(ctx, "DataPlane", cfg.DevelopmentMode).Info(
-				"creating index",
-				"indexField", index.KongPluginInstallationsIndex,
-			)
-			if err := index.KongPluginInstallationsOnDataPlane(ctx, mgr.GetCache()); err != nil {
-				return fmt.Errorf("failed to setup index for KongPluginInstallations on DataPlane: %w", err)
-			}
+			indexOptions = append(indexOptions, index.OptionsForDataPlaneWithKongPluginInstallation()...)
 		}
-		if cfg.KonnectControllersEnabled {
-			if err := index.ExtendableOnKonnectExtension(ctx, mgr.GetCache(), &operatorv1beta1.DataPlane{}); err != nil {
-				return fmt.Errorf("failed to setup index for DataPlanes on KonnectExtensions: %w", err)
-			}
-			if err := index.ExtendableOnKonnectExtension(ctx, mgr.GetCache(), &operatorv1beta1.ControlPlane{}); err != nil {
-				return fmt.Errorf("failed to setup index for ControlPlanes on KonnectExtensions: %w", err)
-			}
-			if err := index.ExtendableOnKonnectExtension(ctx, mgr.GetCache(), &operatorv1beta1.GatewayConfiguration{}); err != nil {
-				return fmt.Errorf("failed to setup index for ControlPlanes on KonnectExtensions: %w", err)
-			}
+	}
+	if cfg.KonnectControllersEnabled {
+		cl := mgr.GetClient()
+		indexOptions = slices.Concat(indexOptions,
+			index.OptionsForExtendables(),
```

Comment:
I think the PR should include some comments about what is the `OptionsForExtendables` used for.

Reply:
PTAL at https://github.com/Kong/gateway-operator/compare/a6d4244a85e207c6fb5cd491a0cc424be98db329..459d80229b863ab93be00260dac19d7185f66b15